File: ros_ws/src/src/intro_pkg1/src/Publisher1.cpp
// Include header files
#include "ros/ros.h"
#include "intro_pkg1/Equ.h"
#include "math.h"

int main(int argc, char **argv) {
    // Initialize the node
    ros::init(argc, argv, "Msg_Publisher");
    // Create a node handler
    ros::NodeHandle nodeHandler;
    // Publisher object. Topic name: /equation
    ros::Publisher publisherObject = nodeHandler.advertise<intro_pkg1::Equ>("equation", 2);
    // Rate controller. 2 times a second, 2 Hz
    ros::Rate rateController = ros::Rate(2);
    // Main loop
    while(ros::ok()) {
        // Message object
        intro_pkg1::Equ msg;
        msg.info.assign("Value of PI");
        msg.value = M_PI;
        // Publish the message
        publisherObject.publish(msg);
        // Delay for handling rate
        rateController.sleep();
    }
    return 0;
}

File: ros_ws/src/src/intro_pkg1/scripts/Publisher1.py
#!/usr/bin/env python
import rospy
from intro_pkg1.msg import Equ
import math

def main():
    # Initialize the node
    rospy.init_node("MSG_Publisher", anonymous=True)
    # Publisher
    publisherObject = rospy.Publisher("equation", Equ, queue_size=2)
    # Rate handler. 2 Hz
    rateHandler = rospy.Rate(2)
    # Loop
    while(not rospy.is_shutdown()):
        # Message
        msg = Equ()
        msg.info = "PI"
        msg.value = math.pi
        # publish message
        publisherObject.publish(msg)
        # Delay section
        rateHandler.sleep()

if __name__ == "__main__":
    try:
        main()
    except rospy.ROSInterruptException:
        rospy.logfatal("Node crashed")
